### ISSUE TITLE
Problem serializing byte 

### ISSUE THREAD

**Issue by karlguillotte** [SphtKr/JSONAPI.NET#64]
Hi, 

Sorry for posting another issue. I just created another `Class/Controller/DbSet` for `Offices`. 

```c#
public class Office {

    [Key]
    public byte Id { get; set; }

    public string Name { get; set; }

    public string City { get; set; }
}
```

Fetching my new endpoint with `Accept: application/vnd.api+json` returns `NOIDCOMPUTABLE!` as ids and valid numbers with `Accept: application/json`. I looked at the tests and it seems that `byte` is supported. I am not sure why the ids are not serialized properly. 

Thanks!

**Comment by csantero** [comment]
If you take a look at `JsonApiFormatter.GetValueForIdProperty()` you'll see the problem:

```
        protected string GetValueForIdProperty(PropertyInfo idprop, object obj)
        {
            if (idprop != null)
            {
                if (idprop.PropertyType == typeof(string))
                    return (string)idprop.GetValue(obj, null);
                if (idprop.PropertyType == typeof(Guid))
                    return idprop.GetValue(obj).ToString();
                if (idprop.PropertyType == typeof(int))
                    return ((int)idprop.GetValue(obj, null)).ToString();
            }
            return "NOIDCOMPUTABLE!";
        }
```

So yes, `byte` is supported for attributes in general, but not as an ID attribute. There's no reason we couldn't support any type though. If you'd like to get involved with the project this would make a great PR!

**Comment by karlguillotte** [comment]
Thanks @csantero. That makes sense. I would love to contribute as I will be using this library in my project. I forked and cloned the repo. I opened the solution in VS. Sorry, for my dumb question here, but I get this error:  

JSONAPI.Json.JsonApiFormatter' does not contain a definition for 'CreateJsonWriter' and no extension method 'CreateJsonWriter' accepting a first argument of type 'JSONAPI.Json.JsonApiFormatter' could be found (are you missing a using directive or an assembly reference?)

I looked at the project references and I have Json.NET assembly checked. Also, Json.NET NuGet package is there when I open the "Manage NuGet Packages" window. Any idea how to solve this problem? 

I have a similar issue with 'System.Web.Http.HttpError' does not contain a definition for 'StackTrace'

**Comment by csantero** [comment]
@karlguillotte It sounds like you don't have the dependencies installed. Do you have package restore enabled for NuGet in Visual Studio? Tools -> Options -> NuGet Package Manager -> General -> Package Restore, and then check both checkboxes.

If you already have that checked, and you have all the deps, then I'm not sure what the problem could be. I just tried a fresh clone of the repo, and hit Built Solution, and it built fine.

**Comment by karlguillotte** [comment]
All checkboxes were checked. It is working since I opened that window and click the Ok button.

**Comment by karlguillotte** [comment]
SphtKr/JSONAPI.NET/pull/66

**Comment by karlguillotte** [comment]
Resolved with #66
